### PR TITLE
Support Non-Basic Auth for Token Endpoint and Custom Claim Requests

### PR DESCRIPTION
**Pull request by lleadbet**
Adding this PR, since some providers (specifically Twitch) require use of `:request_body` from the `oauth2` gem instead of basic as well as explicit claims.

This PR: 

- Adds support for non-basic requests for those endpoints that require it
- Adds an optional claims configuration that allows users to pass the JSON claims object directly through to ensure it's available.

### CODE REVIEW

File: config/settings.yml [hunk 1]
```diff
@@ -29,3 +29,5 @@ plugins:
     default: ''
     type: list
     list_type: compact
+  openid_connect_claims:
+    default: ''
```
Comment (davidtaylorhq) [inline config/settings.yml:33]: Please can you add a trailing newline here

**Review by davidtaylorhq — CHANGES_REQUESTED** [review]
In general this change looks good. Please can you add some RSpec tests to ensure that the new functionality doesn't regress in future?

File: config/settings.yml (hunk 1 above)
Comment (lleadbet) [reply config/settings.yml]: Yep, done!

**Comment by lleadbet** [comment]
Added unit tests; let me know if you have any more feedback!

**Comment by lleadbet** [comment]
@davidtaylorhq - Just a friendly nudge here. :) Our instance is working off my fork, which is awesome, but would love to be part of the mainline.

File: config/locales/server.en.yml [hunk 2]
```diff
@@ -13,6 +13,7 @@ en:
     openid_connect_verbose_logging: "Log detailed openid-connect authentication information to `/logs`. Keep this disabled during normal use."
     openid_connect_authorize_parameters: "URL parameters which will be included in the redirect from /auth/oidc to the IDP's authorize endpoint"
     openid_connect_overrides_email: "On every login, override the user's email using the openid-connect value"
+    openid_connect_claims: "Explicitly define the claims for use with providers that don't pass data back based on scopes."
```
Comment (davidtaylorhq) [inline config/locales/server.en.yml:16]:
```suggestion
    openid_connect_claims: "Explicitly define the claims for use with providers that don't pass data back based on scopes. (JSON)"
```

File: config/settings.yml (hunk 1 above)
Comment (davidtaylorhq) [inline config/settings.yml:33]:
```suggestion
    default: ''
    textarea: true
```
This will make it a little easier to type long JSON strings

**Review by davidtaylorhq — CHANGES_REQUESTED** [review]
This looks great, thanks @lleadbet - just a couple of tiny site-setting-related suggestions, and then it should be good to merge.

**Comment by lleadbet** [comment]
@davidtaylorhq - Done!

**Pull request merged**